### PR TITLE
Add ipytest

### PR DESCRIPTION
**Pull request by pmgrafe**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/ipytest/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install {{ fn }} --no-deps --ignore-installed -vv "
+
+requirements:
+  build:
+    - python
+    - pip
+  run:
+    - python
```
Comment (chrisburr) [inline recipes/ipytest/meta.yaml]: Same:
```suggestion
    - python >=3
```

File: recipes/ipytest/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  sha256: c25e09654b2b431047ab7f381333188be2e8fc66146f1f2aed44dc105128b074
+  
+build:
+  number: 0
+  noarch: python
+  script: "{{ PYTHON }} -m pip install {{ fn }} --no-deps --ignore-installed -vv "
+
+requirements:
+  build:
+    - python
```
Comment (chrisburr) [inline recipes/ipytest/meta.yaml]: Looking at the [build log](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=31170) (line 357 of "run docker build" on linux) it looks like this is Python 3 only?
```suggestion
    - python >=3
```

File: recipes/ipytest/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - pip
+  run:
+    - python
+
```
Comment (chrisburr) [inline recipes/ipytest/meta.yaml]: Does this not depend on `ipython` and `pytest`?
```suggestion
 - ipython
 - pytest
 
```

File: recipes/ipytest/meta.yaml (hunk 2 above)
Comment (pmgrafe) [reply recipes/ipytest/meta.yaml]: In PyPI, it's listed as python 2 or 3, but indeed "import ipytest" does fail for Python 2.7.  I'll go ahead and let the developer know that the package doesn't work with python 2.7 so he can either fix it or remove support.

File: recipes/ipytest/meta.yaml (hunk 3 above)
Comment (pmgrafe) [reply recipes/ipytest/meta.yaml]: Pip installing the package doesn't install ipython, but it does seem like the package is a bit useless without it.

Comment (pmgrafe) [reply recipes/ipytest/meta.yaml]: pytest is also not entirely a requirement. This package can be used to run unit tests using the built-in python unit test runner, but that functionality is deprecated.

Comment (pmgrafe) [reply recipes/ipytest/meta.yaml]: Given that usage without pytest is deprecated and the package fails if pytest is included but not ipython, it makes sense to add both dependencies

**Pull request merged**